e2e-tests/steps/reports.ts

```diff
@@ -0,0 +1,72 @@
+import { Page, expect } from '@playwright/test'
+import Excel, { CellValue } from 'exceljs'
+
+const reportTypeMetaData = {
+  submittedApplications: {
+    columnNames: [
+      'eventId',
+      'applicationId',
+      'personCrn',
+      'personNoms',
+      'referringPrisonCode',
+      'preferredAreas',
+      'hdcEligibilityDate',
+      'conditionalReleaseDate',
+      'submittedAt',
+      'submittedBy',
+      'createdAt',
+    ],
+    callToAction: "Download 'Submitted applications' report",
+  },
+  applicationStatusUpdates: {
+    columnNames: ['eventId', 'applicationId', 'personCrn', 'personNoms', 'updatedBy', 'updatedAt', 'newStatus'],
+    callToAction: "Download 'Application status updates' report",
+  },
+  unsubmittedApplications: {
+    columnNames: ['applicationId', 'personCrn', 'personNoms', 'startedBy', 'startedAt'],
+    callToAction: "Download 'Un-submitted applications' report",
+  },
+}
+
+type ReportType = keyof typeof reportTypeMetaData
+
+export const signInAsMI = async (page: Page, miUser: { name?: string; username: string; password: string }) => {
+  await page.getByLabel('Username').fill(miUser.username)
+  await page.getByLabel('Password').fill(miUser.password)
+  await page.getByRole('button', { name: 'Sign in' }).click()
+}
+
+export const skipForNow = async (page: Page) => {
+  await page.getByRole('button', { name: 'Skip for now' }).click()
+  await expect(page.locator('h1')).toContainText('CAS2: Short-term accommodation')
+}
+
+export const manageInformationReports = async (page: Page) => {
+  await page.getByRole('link', { name: 'Management information reports' }).click()
+  await expect(page.locator('h1')).toContainText('Management information reports')
+}
+
+export const downloadReport = async (reportType: ReportType, page: Page) => {
+  const downloadPromise = page.waitForEvent('download')
+  await page.getByRole('button', { name: reportTypeMetaData[reportType].callToAction }).click()
+  const download = await downloadPromise
+  const path = await download.path()
+  return path
+}
+
+export const confirmColumnNames = async (reportType: ReportType, path: string) => {
+  const workbook = new Excel.Workbook()
+
+  await workbook.xlsx.readFile(path).then(() => {
+    const sh = workbook.getWorksheet('Sheet0')
+
+    const headerCells: CellValue[] = []
+    sh.getRow(1).eachCell(cell => headerCells.push(cell.value))
+
+    expect(headerCells.length === reportTypeMetaData[reportType].columnNames.length).toBe(true)
```

Comment:
This assertion means that if the report is updated with a new column on the API without an update on the FE this test will fail. I was in two minds about this scenario blocking the FE pipeline, but let me know your thoughts.

Reply:
Is this in place to make sure we're not leaking data unexpectedly, or to make sure we keep our E2E tests up to date? If it's the latter then I'm not sure the best way to control that is with tight tests, at the expense of blocking the FE pipeline as you say. Not a major deal though.